src/main/drivers/bus_spi.h

```diff
@@ -59,7 +59,7 @@ typedef enum {
 #elif defined(STM32F7)
     SPI_CLOCK_SLOW          = 256, //00.42188 MHz
     SPI_CLOCK_STANDARD      = 16,  //06.57500 MHz
-    SPI_CLOCK_FAST          = 8,   //13.50000 MHz
+    SPI_CLOCK_FAST          = 4,   //27.00000 MHz
```

Comment:
It is not appropriate to change this. Other devices use this enumeration besides SPI flash devices.

Reply:
This changes that definition on F7 to be in line with other processors. F4 @ 21MHz, H7 @ 25MHz, G4 @21.25MHz. This speed is set per device. As device that's good at 25MHz on one processor should be run at approx the same speed on all processors.

Reply:
Bar FLASH, the only devices using SPI_CLOCK_FAST are:

| Device | Max. SPI Clock | Reference |
|-|-|-|
| MPU6000 | 1 MHz | https://invensense.tdk.com/wp-content/uploads/2015/02/MPU-6000-Datasheet1.pdf
| MPU6500 | 20 MHz | http://invensense.tdk.com/wp-content/uploads/2020/06/PS-MPU-6500A-01-v1.3.pdf |
| MPU 9250 | 20 MHz  | https://invensense.tdk.com/wp-content/uploads/2015/02/PS-MPU-9250A-01-v1.1.pdf |

We are seriously overclocking the MPU6000 it would appear, and on other processors pushing the others too. Perhaps on the F4 it was decided that 27MHz was overcooking things just a bit too much.

There's evidently been some struggling with such things before.

In `flashSpiInit()` for example.

```
    //Maximum speed for standard READ command is 20mHz, other commands tolerate 25mHz
    //spiSetDivisor(busdev->busdev_u.spi.instance, SPI_CLOCK_FAST);
    spiSetDivisor(busdev->busdev_u.spi.instance, SPI_CLOCK_STANDARD*2);
```

On an F7 this is only 3.38MHz. My code leaves this as it whilst the chip ID is read and then overrides it to a higher speed.

In all cases bar the Winbond W25Q16 (good for 50MHz on read) and Micron M25P16 (20MHz) this is set to SPI_CLOCK_ULTRAFAST, and in those cases to SPI_CLOCK_FAST. We should be good to set the W25Q16 to SPI_CLOCK_ULTRAFAST.

I'll revert this change, so the only device to suffer will be the M25P16 and that will still be running 4 times as fast as before.

Reply:
It is well known that the MPU6000 datasheet is lacking information about the SPI bus speed. It's undocumented but it actually supports high speed SPI for the core data registers (ACC, gyro, temp) but only slow(er) SPI for the other general purpose registers. The point is that you cannot change a general enumeration used by other devices without certifying that all devices support this change And MPU6000 won't support 27MHz.

Reply:
Is running FLASH rated at 50MHz over it's full temperature range at 54MHz too risky would you say? Is so it would be good to adjust the F7's SPI clock down to give us 50MHz. Any idea why it's set differently to the F4?

Reply:
By changing the definition of PLL_N in src/main/startup/system_stm32f7xx.c I've been able to change the SPI_CLOCK_ULTRAFAST speed on the F7 from 54MhZ down to 50MHz, as on the H7 which means we wouldn't have to overclock the FLASH to avoid running it at only 27MHz. I tried changing the PLLI2S.PLLI2SN value from 216 to 200 (which seems more targeted), but that had no effect.

Reply:
The following suggests this is worthy of some investigation.

```
  /// TODO: F7 check if this is the best configuration for the clocks.
  // current settings are just a copy from one of the example projects
```

Reply:
Our initialisation appears to come from https://www.st.com/resource/en/user_manual/dm00189702-description-of-stm32f7-hal-and-lowlayer-drivers-stmicroelectronics.pdf

Reply:
@etracer65 The H7 sets SPI_CLOCK_FAST to 25MHz. Is this too fast for the MPU6000 too? If so this should also be changed, but perhaps not by this PR.

Reply:
Now I'm not overclocking anything beyond the 20MHz you mention.